### PR TITLE
[dashboards] Add new Dashboard API docs

### PR DESCRIPTION
**Pull request by enbashi**
### What does this PR do?
Adds documentation and code examples for the new `dashboards` API.

### Motivation
New API.

### Preview link
https://docs-staging.datadoghq.com/rami/add_new_dashboard_api/api/#dashboards

### Additional Notes
- This new API replaces the `timeboards` API and soon will also replace `screenboards` API. 
- Python and Ruby code examples are based on the new API which are not released yet. See https://github.com/DataDog/datadogpy/pull/351 and https://github.com/DataDog/dogapi-rb/pull/167

### CODE REVIEW

File: content/api/dashboards/code_snippets/result.api-dashboard-get.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "dash": {
```
Comment (MLaureB) [inline content/api/dashboards/code_snippets/result.api-dashboard-get.sh]: This is an example for the legacy timeboard API: we need to change it with the new one 🙂

**Comment by l0k0ms** [comment]
@MLaureB @enbashi Should we remove the timeboard API doc then since this one replaces it?

**Comment by MLaureB** [comment]
@l0k0ms , so we should not remove the timeboard API doc completely as we're gonna still have a bunch of people using it who might want to refer to it, but would that be possible to only show the new `Dashboards` one (and hide the `Timeboards` one) and put somewhere a link to the legacy timeboards API, in case people still want to access it?

File: content/api/dashboards/code_snippets/result.api-dashboard-get.sh (hunk 1 above)
Comment (enbashi) [reply content/api/dashboards/code_snippets/result.api-dashboard-get.sh]: oops. Thanks for catching that.

**Comment by l0k0ms** [comment]
@DataDog/baklava for visibility: this PR now removes the current Timeboard API endpoint doc and replaces it with the new Dashboard endpoint. 

It removes the current graphing/dashboard/widget/page and put it into the FAQ section for now. as a freeze.

Next steps are: 

1 - Create a new branch to re-introduce the timeboard API as a stand-alone page in the FAQ section in order to keep documentation available in case customers would like to refer to it. 

2 - Create a new branch to create a whole widget sub-section in the graphing section that will directly implement the new widget logic, and will be an update on the current widget doc.

**Comment by l0k0ms** [comment]
PR to introduce timeboard API as a FAQ: https://github.com/DataDog/documentation/pull/3950 ping @MLaureB

**Comment by l0k0ms** [comment]
PR to introduce the new widget section: https://github.com/DataDog/documentation/pull/3948

**Pull request merged**